src/components/dropdown/dropdown.ts

```diff
@@ -0,0 +1,579 @@
+import { LitElement, html } from 'lit';
+import { property, query, queryAssignedElements } from 'lit/decorators.js';
+import { Constructor } from '../common/mixins/constructor';
+import { EventEmitterMixin } from '../common/mixins/event-emitter';
+import { styles } from './dropdown.material.css';
+import { watch } from '../common/decorators';
+import {
+  IgcPlacement,
+  IgcToggleEventMap,
+  IToggleOptions,
+} from '../toggle/utilities';
+import IgcDropDownItemComponent from './dropdown-item';
+import { IgcToggleController } from '../toggle/toggle.controller';
+import IgcDropDownGroupComponent from './dropdown-group';
+
+export enum DropDownActionKey {
+  ESCAPE = 'escape',
+  ENTER = 'enter',
+}
+
+export interface IgcDropDownEventMap extends IgcToggleEventMap {
+  igcChange: CustomEvent<ISelectionChangeEventArgs>;
+}
+
+export interface ISelectionChangeEventArgs {
+  newItem: IgcDropDownItemComponent | null;
+}
+
+/**
+ * Represents a DropDown component.
+ *
+ * @element igc-dropdown
+ *
+ * @fires igcChange - Emitted when the selected item changes.
+ *
+ * @slot target - Renders the dropdown's target element.
+ * @slot - Renders the dropdown list items.
+ *
+ *
+ */
+export default class IgcDropDownComponent extends EventEmitterMixin<
+  IgcDropDownEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  public static readonly tagName = 'igc-dropdown';
+
+  public static styles = styles;
+
+  private toggleController!: IgcToggleController;
+  private selectedItem!: IgcDropDownItemComponent | null;
+  private activeItem!: IgcDropDownItemComponent;
+  private target!: HTMLElement;
+
+  private get allItems(): IgcDropDownItemComponent[] {
+    const groupItems: IgcDropDownItemComponent[] = this.groups.flatMap(
+      (group) => group.items
+    );
+    return [...this.items, ...groupItems];
+  }
+
+  @query('#igcDDLContent')
+  protected content!: HTMLElement;
+
+  @query('#igcScrollContainer')
+  protected scrollContainer!: HTMLElement;
+
+  @queryAssignedElements({ flatten: true, selector: 'igc-dropdown-item' })
+  protected items!: Array<IgcDropDownItemComponent>;
+
+  @queryAssignedElements({ flatten: true, selector: 'igc-dropdown-group' })
+  protected groups!: Array<IgcDropDownGroupComponent>;
+
+  @queryAssignedElements({ slot: 'target' })
+  private targetNodes!: Array<HTMLElement>;
+
+  /** Sets the open state of the dropdown list. */
+  @property({ type: Boolean })
+  public open = false;
+
+  /** Sets the dropdown list's positioning strategy. */
+  @property({ attribute: false })
+  public positionStrategy: 'absolute' | 'fixed' = 'absolute';
+
+  /** The preferred placement of the dropdown list around the target element. */
+  @property({ attribute: false })
```

Comment:
Since we are using union type we should provide the options in the API comment like this `@type {"top" | "bottom" | ...}`